es/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java

```diff
@@ -237,15 +249,13 @@ static S3ClientSettings getClientSettings(final Settings settings, final String
     static S3ClientSettings getClientSettings(final RepositoryMetaData metadata, final AWSCredentials credentials) {
         final Settings.Builder builder = Settings.builder();
         for (final String key : metadata.settings().keySet()) {
-            builder.put(PREFIX + "provided" + "." + key, metadata.settings().get(key));
+            builder.put(DEFAULT_PREFIX + key, metadata.settings().get(key));
         }
-        return getClientSettings(builder.build(), "provided", credentials);
+        return getClientSettings(builder.build(), "", credentials);
```

Comment:
we could remove `clientName` argument everywhere as it is not used anymore

Reply:
I have another commit almost ready with further cleanups (it includes most your comments), coming up in a bit